mlflow/llama_index/__init__.py

```diff
@@ -348,32 +354,158 @@ def _load_pyfunc(path, model_config: Optional[Dict[str, Any]] = None):
 
 
 class _LlamaIndexModelWrapper:
```

Comment:
Random idea: I found almost all lines in class have conditioned by engine type. What about having subclasses like `_LlamaIndexChatEngineWrapper`?

Reply:
Genius